src/app/caesar-portal/group-list/create-edit-dialog/create-edit-dialog.component.ts

```diff
@@ -0,0 +1,287 @@
+    private profileService: ProfileService,
+    private updateGroupService: UpdateGroupService,
+    private datePipe: DatePipe
+  ) { }
+
+  ngOnInit() {
+    this.dateStart = new Date();
+    this.dateFinish = new Date();
+    this.getCurrentUserInfo();
+    this.showAllDirections();
+    this.showAllBudgetOwners();
+    this.showAllStages();
+  }
+
+  ngOnChanges() {
+  }
+
+  public onTeachersChanged(teachers: {id: number}[]): void {
+    this.teachers = teachers;
+  }
+
+  public onExpertsChanged(experts: string[]): void {
+    this.experts = experts;
+  }
+
+  private getCurrentUserInfo(): void {
+    this.profileService.getCurrentUser()
+      .subscribe(
+        (data: Response) => {
+          this.currentUser = data.json();
+          this.selectedLocation = this.currentUser.location;
+          this.showAllLocations();
+        },
+        (error) => {
+          this.error = error;
+          console.error(error);
+        }
+      );
+  }
+
+  private showAllLocations() {
+    this.updateGroupService.getLocations().subscribe(
+      (data: Location[]) => {
+        this.locations = data;
+        if (typeof this.isEditMode !== 'undefined') {
+          if (!this.isEditMode) {
+            this.generateName();
+          }
+        }
+      },
+      error => console.log(error));
+  }
+
+  private showAllDirections() {
+    this.updateGroupService.getDirections().subscribe(
+      (data: Direction[]) => {
+        this.directions = data;
+        this.selectedDirection = this.directions[0];
+        if (typeof this.isEditMode !== 'undefined') {
+          if (!this.isEditMode) {
+            this.generateName();
+          }
+        }
+      },
+      error => console.log(error));
+  }
+
+  private showAllBudgetOwners() {
+    this.updateGroupService.getBudgetOwners().subscribe(
+      (data: BudgetOwner[]) => {
+        this.budgetOwners = data;
+        this.selectedBudgetOwner = this.budgetOwners[0];
+      },
+      error => console.log(error));
+  }
+
+  private showAllStages() {
+    this.updateGroupService.getStages().subscribe(
+      (data: Stage[]) => {
+        this.stages = data;
+        this.selectedStage = this.stages[0];
+      },
+      error => console.log(error));
+  }
+
+  public changeSelectedLocation(location) {
+    this.selectedLocation = location;
+    this.generateName();
+  }
+
+  public changeSelectedDirection(direction) {
+    this.selectedDirection = direction;
+    this.generateName();
+    this.calculateFinishDate();
+  }
+
+  public changeSelectedStage(stage) {
+    this.selectedStage = stage;
+  }
+
+  public changeSelectedBudgetOwner(budgetOwner) {
+    this.selectedBudgetOwner = budgetOwner;
+  }
+
+  public isActive(budgetOwner) {
+    return this.selectedBudgetOwner === budgetOwner;
+  }
+
+  // autogeneration group's name
+
+  public generateName () {
+    let result = '';
+    if (this.selectedLocation) {
+      result += `${this.locationNamesAlias[this.selectedLocation.id]}-`;
```

Comment:
since you're already using `` strings you can add both in one line as variables in it (Ie `${some_data}-${some_other_data}`), even if it has function call in it